test/test_enzyme.jl

```diff
@@ -169,7 +167,10 @@ end
     topology = (Periodic, Periodic, Bounded)
 
     grid = RectilinearGrid(size=(Nx, Ny, Nz); x, y, z, topology)
-    diffusion = VerticalScalarDiffusivity(κ=0.1)
+    κ = ZFaceField(grid)
+    fill!(κ, 0.1)
+
+    diffusion = VerticalScalarDiffusivity(; κ)
```

Comment:
Please restore this code to how it was!

Reply:
I think it would be better to mark this as `@test_broken` then. We need the constructor to be type stable to do this work!

Reply:
Please understand what I am saying: the point of this test is not to get a numerical result. It's supposed to verify that we can use this method for doing parameter estimation with this specific setup. So if the code breaks then the test fails.

Reply:
ok I can do that

Reply:
> There was a problem with this test. The constructor of the ScalarDiffusivity is not type-stable (like most constructors)

Many constructors are type stable, and we have to work to make them all type stable to support AD.

This test passed previously, so not sure what changed in this PR?